### PR TITLE
Project init!

### PR DESCRIPTION
**Pull request by JPLachance**
This is it, our simple lib of react component! Clone it, build it, beat it, I want your feedback!

### CODE REVIEW

File: docs/Index.tsx [hunk 1]
```diff
@@ -0,0 +1,371 @@
+import {Model as BackboneModel} from 'backbone';
```
Comment (JPLachance) [inline docs/Index.tsx]: Yes, that file is a big blob of code. For now, we only need a place where to put code to test our components, but we will eventually created a structured project for component doc.

File: docs/index.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<html lang="en">
+<head>
+    <title>React Vapor</title>
+
+    <meta charset="UTF-8">
+</head>
+<body class="coveo-styleguide">
+<div class="header flex" style="min-height: 65px;">
+    <div class="header-section h1 p2 flex flex-center">
+        Will it blend?
```
Comment (wfortin) [inline docs/index.html]: react-vapor as a title ?

File: package.json [hunk 3]
```diff
@@ -1,19 +1,79 @@
...
   "repository": {
     "type": "git",
     "url": "git+https://github.com/JPLachance/react-vapor.git"
   },
+  "scripts": {
+    "postinstall": "typings install",
+    "start": "gulp docs && webpack-dev-server --devtool eval --inline --hot --progress --colors",
+    "build": "gulp",
+    "prepublish": "npm run build",
+    "add-file": "tsconfig . -i 2"
```
Comment (wfortin) [inline package.json]: Is that needed since you added this to the gulpfile ?

File: tslint.json [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    "whitespace": [
+      true,
+      "check-branch",
+      "check-decl",
+      "check-operator",
+      "check-separator",
+      "check-type"
+    ]
+  }
+}
```
Comment (wfortin) [inline tslint.json]: PrettyTypescript has its own TSLint file, so maybe we shouldn't have this.

File: gulpfile.js [hunk 5]
```diff
@@ -1,3 +1,137 @@
...
+});
+//</editor-fold>
+
+//<editor-fold desc="Typescript d.ts generation">
+gulp.task('internalDefs', false, () => {
+  return require('dts-generator').default({
+    name: 'ReactVapor',
+    project: './',
+    baseDir: './src/',
+    out: 'dist/ReactVapor.d.ts',
```
Comment (JPLachance) [inline gulpfile.js]: Already done, hit F5!

File: lib/react-tether.d.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    style?: Object;
+    onUpdate?: Function;
+    onRepositioned?: Function;
+    children?: [React.ReactNode, boolean | React.ReactNode];
+  }
+}
+
+declare module 'react-tether' {
+  export = TetherComponent;
+}
```
Comment (JPLachance) [inline lib/react-tether.d.ts]: Yes, when I'll have a little bit of time for it!

File: tslint.json (hunk 4 above)
Comment (JPLachance) [inline tslint.json]: Yes, but it requires contributors to know that and setup their IDE V.S. the default tslint file. Yes it's duplicated, but I think it's a good think to simplify devs life.

Comment (wfortin) [inline tslint.json]: Ok so it's the same config ?

Comment (JPLachance) [inline tslint.json]: Yes, I copy/pasted it.

File: docs/Index.tsx [hunk 7]
```diff
@@ -0,0 +1,371 @@
...
+const onClickCancel = (modelCid: string, email: string, sendEmail: boolean) => {
+  return {
+    type: actions.CancelMemberEdition,
+    modelCid,
+    email,
+    sendEmail
+  };
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
```
Comment (sroy3) [inline docs/Index.tsx]: If you want to type this, ownProps has `IMemberEditViewPropsConnected` as its type and dispatch has `(action: Redux.Action) => void` as its type.

File: docs/index.html (hunk 2 above)
Comment (JPLachance) [inline docs/index.html]: Done!

File: package.json (hunk 3 above)
Comment (JPLachance) [inline package.json]: Removed!

File: docs/Index.tsx (hunk 7 above)
Comment (JPLachance) [inline docs/Index.tsx]: Done! Thanks ;)

**Pull request merged**